src/Extension/Console/ExtensionDisableCommand.php

```diff
@@ -97,27 +97,41 @@ protected function allEnabledExtensions()
     protected function getExtensionsToDisable()
     {
         $extensionIds = [];
-        if ($this->input->getOption('all')) {
-            // All Extensions
-            $extensionIds = $this->allEnabledExtensions();
-        } elseif ($this->input->getOption('all-except-bundled')) {
-            // All exc
-            $extensionIds = array_filter($this->allEnabledExtensions(), function ($extension) {
-                return substr($extension, 0, 6) !== 'flarum' ?: $this->info("Extension: $extension is bundled, ignoring") && false;
-            });
+
+        if (empty($extensions)) {
+            $this->info('No extensions to process.');
         }
 
-        // Process user-specified extensions
         foreach ($this->input->getArgument('extensions') as $extension) {
             if (! $this->extensions->getExtension($extension)) {
-                $this->error("Extension: $extension is not installed.");
+                $this->error("Extension: $extension is not installed, ignoring.");
             } elseif (! in_array($extension, $extensionIds)) {
                 $extensionIds[] = $extension;
             }
         }
 
+        if ($this->input->getOption('all')) {
+            $extensionIds = $this->allEnabledExtensions();
+        } elseif ($this->input->getOption('all-except-bundled')) {
+            $extensionIds = array_merge($extensionIds, array_filter($this->allEnabledExtensions(), function ($extension) use ($extensionIds) {
+                if (substr($extension, 0, 6) === 'flarum' && ! in_array($extension, $extensionIds)) {
```

Comment:
I would suggest adding an `$onlyCore` parameter to `$this->allEnabledExtensions()` and doing the check there against the actual vendor.

I think there's a good chance to find vendor names that start with "flarum" or even "flarum-" in the future. We already have https://packagist.org/packages/flarumite/

Not sure how that would play with the info message though.

Reply:
Do we somewhere store a list of core extensions? I'm not sure how we could really validate this.

Reply:
I'd say a core extension is an extension whose vendor is `flarum`. And this can't be checked against the extension ID because we lose the separation between vendor and package. This should be checked against the extension object.

Reply:
I'm not sure how that fixes the issue though: if an extension's vendor is `flarum`, it's ID will start with `flarum-`, and vice versa.

Reply:
The problem is that `flarum-something` is a valid vendor name that someone could register or even only use locally, and there's no way to tell it apart from the actual `flarum` vendor if we are not reading the `vendor` attribute of the extension.

Why we remove the separation between vendor and package in the extension ID would be a question for another issue... Keeping it would give a bit more flexibility for situations like these and completely avoid the possibility for two extensions having the same ID

Reply:
That's a good point, I hadn't thought of that.